test/verify/check-storage-swap

```diff
@@ -24,27 +24,94 @@ import testlib
+        self.click_card_dropdown("Solid State Drive", "Create partition table")
+        self.confirm()
+        b.wait_text(self.card_row_col("GPT partitions", 1, 1), "Free space")
+        self.click_dropdown(self.card_row("GPT partitions", 1), "Create partition")
+        self.dialog({"type": "swap"})
+        b.wait_text(self.card_row_col("GPT partitions", 1, 2), "Swap")
 
-        b.wait_visible(self.card("Unformatted data"))
-        m.execute(f"mkswap {disk}")
-        b.wait_visible(self.card("Swap"))
+        # It should have been started and have a fstab entry
+        self.click_card_row("GPT partitions", 1)
+        b.wait_text(self.card_desc("Swap", "Used"), "0")
+        self.assertIn("defaults", m.execute(f"findmnt --fstab -n -o OPTIONS {disk}1"))
+
+        # Stopping should set it to noauto
+        b.click(self.card_button("Swap", "Stop"))
         b.wait_text(self.card_desc("Swap", "Used"), "-")
+        self.assertIn("noauto", m.execute(f"findmnt --fstab -n -o OPTIONS {disk}1"))
 
+        # Start it again to test teardown below
         b.click(self.card_button("Swap", "Start"))
         b.wait_text(self.card_desc("Swap", "Used"), "0")
+        self.assertIn("defaults", m.execute(f"findmnt --fstab -n -o OPTIONS {disk}1"))
 
-        self.assertEqual(m.execute(f"lsblk -n -o MOUNTPOINT {disk}").strip(), "[SWAP]")
+        # It should have the right partition type
+        b.wait_visible(self.card("Swap"))
+        b.wait_text(self.card_desc("Partition", "Type"), "Linux swap space")
 
-        b.click(self.card_button("Swap", "Stop"))
-        b.wait_text(self.card_desc("Swap", "Used"), "-")
+        # Set it to something else
+        b.click(self.card_desc_action("Partition", "Type"))
+        self.dialog({"type": "0fc63daf-8483-4772-8e79-3d69d8477de4"})
+        b.wait_text(self.card_desc("Partition", "Type"), "Linux filesystem data")
+
+        # Correct it by reformatting
+        self.click_card_dropdown("Swap", "Format")
+        self.dialog_wait_open()
+        self.dialog_set_val("type", "swap")
+        b.wait_in_text("#dialog .modal-footer-teardown", f"{disk}1")
+        b.wait_in_text("#dialog .modal-footer-teardown", "stop, format")
+        self.dialog_apply_secondary()
+        self.dialog_wait_close()
+        b.wait_text(self.card_desc("Partition", "Type"), "Linux swap space")
+
+        # Delete the partition, the fstab entry should disappear
+        self.click_card_dropdown("Partition", "Delete")
+        self.confirm()
+        b.wait_visible(self.card("Solid State Drive"))
+        m.execute(f"! findmnt --fstab -n -o OPTIONS {disk}1")
+
+        # Format as swap on the command line, starting it should add
+        # fstab entry
+        m.execute(f"mkswap -f {disk}")
+        b.click(self.card_button("Swap", "Start"))
+        b.wait_text(self.card_desc("Swap", "Used"), "0")
+        self.assertIn("defaults", m.execute(f"findmnt --fstab -n -o OPTIONS {disk}"))
```

Comment:
This check seems to be a bit racy fails on both [Arch](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-arch-storage/log.html#92-1) and [debian-testing](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240118-085557-ffd6b834-debian-testing-storage/log.html#92-1)

Reply:
Yes, clicking "Start" will first activate the Swap and then add the fstab entry. So it is feasible that the "Used" number appears before the fstab entry. I changed this into a testlib.wait.